lib/noticed/base.rb

```diff
@@ -74,7 +74,13 @@ def run_delivery(recipient, enqueue: true)
 
       # Run database delivery inline first if it exists so other methods have access to the record
       if (index = delivery_methods.find_index { |m| m[:name] == :database })
+        # Remove it from the array of deliveries so it is only executed once
         delivery_method = delivery_methods.delete_at(index)
+
+        # If delayed, run_delivery_method will return the scheduled job instead of the record
+        delivery_method[:options].delete(:delay) if delivery_method[:options].key?(:delay)
```

Comment:
What if we do this as a validation on the DeliveryMethods::Database class instead? That way we can raise an `ArgumentError` with a message about it right there.